webapi/src/main/scala/org/knora/webapi/responders/v1/SipiResponderV1.scala

```diff
@@ -160,15 +160,22 @@ class SipiResponderV1 extends ResponderV1 {
             conversionResultResponse <- recoveredConversionResultFuture
 
             // get file type from Sipi response
-            responseAsMap: Map[String, JsValue] = conversionResultResponse.entity.asString.parseJson.asJsObject.fields.toMap
+            responseAsMap: Map[String, JsValue] = conversionResultResponse.entity.asString.parseJson.asJsObject.fields
 
             statusCode: Int = responseAsMap.getOrElse("status", throw SipiException(message = "Sipi did not return a status code")) match {
                 case JsNumber(ftype: BigDecimal) => ftype.toInt
                 case other => throw SipiException(message = s"Sipi did not return a correct status code, but ${other.toString()}")
             }
 
             // check if Sipi returned a status code != 0
-            _ = if (statusCode != 0) throw BadRequestException(s"Sipi returned a HTTP 200 status code, but an unsuccessful status code ${statusCode}")
+            _ = if (statusCode != 0) {
+                // Sipi returned an unsuccessful status code, get the error message
+                val sipiErrorMessage: String = responseAsMap.getOrElse("message", throw SipiException(message = "Sipi did not return an error message with key 'message' although status is != 0")) match {
+                    case JsString(msg) => msg
+                    case other => throw SipiException(message = s"Sipi did not return an error message of type string, but ${other.toString()}")
+                }
+                throw BadRequestException(s"Sipi returned an HTTP 200 status code, but an unsuccessful status code ${statusCode} with error message: '${sipiErrorMessage}'")
```

Comment:
This doesn't make sense to me:

```
Sipi returned an HTTP 200 status code, but an unsuccessful status code ${statusCode}
```

If the status code was 200, then it was successful, so why do you say it was unsuccessful?